StitchVolumes/StitchVolumes.py

```diff
@@ -0,0 +1,706 @@
+            )
+        self._parameterNode = inputParameterNode
+
+        # Initial GUI update
+        self.updateGUIFromParameterNode()
+
+    def updateGUIFromParameterNode(self, caller=None, event=None):
+        """
+        This method is called whenever parameter node is changed.
+        The module GUI is updated to show the current state of the parameter node.
+        """
+
+        # Disable all sections if no parameter node is selected
+        self.ui.basicCollapsibleButton.enabled = self._parameterNode is not None
+        # self.ui.advancedCollapsibleButton.enabled = self._parameterNode is not None
+        if self._parameterNode is None:
+            return
+
+        # Update each widget from parameter node
+        # Need to temporarily block signals to prevent infinite recursion (MRML node update triggers
+        # GUI update, which triggers MRML node update, which triggers GUI update, ...)
+
+        wasBlocked = self.ui.roiSelector.blockSignals(True)
+        self.ui.roiSelector.setCurrentNode(
+            self._parameterNode.GetNodeReference("StitchedVolumeROI")
+        )
+        self.ui.roiSelector.blockSignals(wasBlocked)
+        wasBlocked = self.ui.volumeSelector1.blockSignals(True)
+        self.ui.volumeSelector1.setCurrentNode(
+            self._parameterNode.GetNodeReference("InputVol1")
+        )
+        self.ui.volumeSelector1.blockSignals(wasBlocked)
+        wasBlocked = self.ui.volumeSelector2.blockSignals(True)
+        self.ui.volumeSelector2.setCurrentNode(
+            self._parameterNode.GetNodeReference("InputVol2")
+        )
+        self.ui.volumeSelector2.blockSignals(wasBlocked)
+        wasBlocked = self.ui.volumeSelector3.blockSignals(True)
+        self.ui.volumeSelector3.setCurrentNode(
+            self._parameterNode.GetNodeReference("InputVol3")
+        )
+        self.ui.volumeSelector3.blockSignals(wasBlocked)
+        wasBlocked = self.ui.volumeSelector4.blockSignals(True)
+        self.ui.volumeSelector4.setCurrentNode(
+            self._parameterNode.GetNodeReference("InputVol4")
+        )
+        self.ui.volumeSelector4.blockSignals(wasBlocked)
+        wasBlocked = self.ui.volumeSelector5.blockSignals(True)
+        self.ui.volumeSelector5.setCurrentNode(
+            self._parameterNode.GetNodeReference("InputVol5")
+        )
+        self.ui.volumeSelector5.blockSignals(wasBlocked)
+
+        # What about other values? (current text, e.g.)?  The example code did not update them here
+
+        # Update buttons states and tooltips
+        # Enable the Stitch Volumes button if there is an ROI, at least two original volumes, and a name for the output vol
+        if (
+            self._parameterNode.GetNodeReference("StitchedVolumeROI")
+            and self._parameterNode.GetNodeReference("InputVol1")
+            and self._parameterNode.GetNodeReference("InputVol2")
+            and self._parameterNode.GetParameter("OutputVolName")
```

Comment:
For user convenience and conform with Slicer conventions, please create an output node selector. You can set the None text to "Create new volume" if you don't want to force the user create/select the output node manually.

Reply:
Changed to node selector for output as suggested